_posts/2022-11-25-prac1b-data-annotation.md

```diff
@@ -0,0 +1,188 @@
+
+At the end of these exercises, you should have 4 annotation CSV files saved in your `OxfordImageBrowser/annotation/me` directory.
+
+<!-- browser_dir = '/Users/<yourName>/OxfordImageBrowser/annotation/me/' -->
+
+Copy the `my-annotations.csv` file over to your `~/practicals/data/` folder.
+
+
+
+<!-- ### Uncodeable Activities
+
+In your resulting time-series file, you might notice that some of your annotations are 'uncodeable'. Here are some ways to fix this.
+
+* Sleep
+
+Do the following if you want your annotated time series to have 'sleep' events.
+
+Open your annotations file at `~/wearable-teaching/practicals/data/my-annotations.csv` and manually change the annotation of the events you believe correspond to your sleeping hours into `7030 sleeping`.
+
+* Other Events
+
+Do the following if you notice that there are other events in your time series which you have annotated but appears to be `uncodeable` in your plot.
+
+Open the file at `~/wearable-teaching/practicals/data/prac3_process_camera/annotation-label-dictionary.csv` and manually append the table.
+
+For example, 'leisure;recreation;outdoor;15533 rock or mountain climbing' is not currently in the annotation-label dictionary. Append the table, putting 'leisure;recreation;outdoor;15533 rock or mountain climbing' in the 'annotation' column, and a category which you feel is appropriate under the 'label:Doherty2018' column (in this case I've assigned the latter as moderate):
+
+![](./assets/figs/add_uncodeable.png) -->
+
+
+
+
+# 5. (Optional) Challenge: Cross-check your own annotations with others
+Try to understand if you can use [this](https://github.com/activityMonitoring/oxford-wearable-camera-browser/blob/master/kappaScoring.py) to compare your annotations with another student. To run this script, you will first need to install four depedencies:
+
+```shell
+$ pip3 install argparse
+$ pip3 install numpy
+$ pip3 install pandas
+$ pip3 install -U scikit-learn
+```
+
+Then generate the list of image file names in [this](https://github.com/activityMonitoring/oxford-wearable-camera-browser/blob/master/training/train1-fileList.txt) format and store this as `cdt-fileList.txt` under the `~/oxford-wearable-camera-browser/training` folder. Move your own annotation to the same place and rename it as `cdt-ref.csv`.
+
+Now go a get another annotation file from your friend, you can run the comparison script via:
+```shell
+$ python kappaScoring.py /..path../annotation.csv
+```
+
+To generate the file list, you can use this Python script:
+```python
+import os 
+import glob
+
+path2gen = '/Users/hangy/plots'
+file_list = glob.glob(os.path.join(path2gen, '*.JPG'))
+
+with open('file.txt', 'w') as f:
+    for item in file_list:
+        txt2write = item.split('/')[-1]
+        print(txt2write)
+        f.write("%s\n" % txt2write)
+```
+When you are done, let your tutor know what your kappa score is.
+
+
+# Matching with accelerometer timings? 
```

Comment:
Is this supposed to be a new section? 
It might help to view the markdown files using a markdown reader for proofreading purpose if it is not easy to render to page locally :D

Reply:
Yeah, this is just in case we can also add a section on how to match up the camera annotations with the accelerometer data.